piecedirectory/piecedirectory.go

```diff
@@ -302,21 +303,31 @@ func (ps *PieceDirectory) addIndexForPieceThrottled(ctx context.Context, pieceCi
 	// Note: Once we start the add index operation we don't want to cancel it
 	// if one of the waiting threads cancels its context. So instead we use the
 	// PieceDirectory's context.
-	op.err = ps.addIndexForPiece(ps.ctx, pieceCid, dealInfo)
+	op.err = ps.addIndexForPiece(ps.ctx, pieceCid, dealInfo, inFile)
 
 	// Return the result
 	log.Debugw("add index: completed", "pieceCid", pieceCid)
 	return op.err
 }
 
-func (ps *PieceDirectory) addIndexForPiece(ctx context.Context, pieceCid cid.Cid, dealInfo model.DealInfo) error {
+func (ps *PieceDirectory) addIndexForPiece(ctx context.Context, pieceCid cid.Cid, dealInfo model.DealInfo, inFile string) error {
 	// Get a reader over the piece data
 	log.Debugw("add index: get index", "pieceCid", pieceCid)
-	reader, err := ps.pieceReader.GetReader(ctx, dealInfo.MinerAddr, dealInfo.SectorID, dealInfo.PieceOffset, dealInfo.PieceLength)
-	log.Debugf("got the piece reader for piece %s and deal %s", pieceCid, dealInfo.DealUuid)
-	if err != nil {
-		return fmt.Errorf("getting reader over piece %s: %w", pieceCid, err)
+	var reader types.SectionReader
+
+	reader, err := os.OpenFile(inFile, os.O_RDONLY, 0)
```

Comment:
This will result in incorrect offset for CIDs. We read from pieceReader because we need exact location in the unsealed copy to get the data back.